tools/ports/__init__.py

```diff
@@ -401,6 +401,7 @@ def add_deps(node):
       dep = ports_by_name[d]
       if dep not in port_set:
         port_set.add(dep)
+        ports_needed.add(dep.name)
```

Comment:
Could you add this line after ilne 500 instead?

Reply:
I don't understand. Line 500 is this:

```python
  for port in dependency_order(needed):
    port.linker_setup(Ports, settings)
    # port.get returns a list of libraries to link
    ret += port.get(Ports, settings, shared)
```

And your original suggestion was: _add ports_needed.add(dep) right after port_set.add(deps) perhaps?_

Reply:
For the record:

* I only made this change because you suggested it although I thought it was not necessary
> We have a if port.needed(settings): check on line 500 which we should perhaps delete.
> **Alternatively you could make port.needed(settings) return true if you add ports_needed.add(dep) right after port_set.add(deps) perhaps?**
> I think perhaps **we should do both** of the above since the check on line 500 seems redundant, but it **would also be nice** if .needed() returned the correct thing for contrib deps
* The call `node.process_dependencies(settings)` on line 397 is potentially modifying dependencies (see `sdl2_mixer.py` for an example), so unfortunately calling `resolve_dependencies` is not without side effects and modify dependencies. Every time it gets called, the `deps` field of the port gets modified and gets longer and longer... So I don't think that `resolve_dependencies` should casually be called to "simply to retrieve the dependency list"

I have now removed this line. So the fix for this issue is to simply remove the test at line 500. Which also makes the code in `add_cflags` and `get_libs` identical in how they deal with dependencies (which makes total sense)